### PR TITLE
feat: added auth functionality

### PR DESCRIPTION
**Pull request by Ankur9669**
Added auth functionality

Files to check:
1. LoginForm.js
2. SignupForm.js
3. loginUser.js
4. signUpUser.js

You can choose not to review the UI files as they are same from ecomm, but please give UI suggestions on which i can work and improve

# Login Page
image

#SignUp Page
image

To navigate to login page please click login button on navbar
image

To navigate to signup page please click **signup with us button** on login page
image

### CODE REVIEW

**Review by Naman-Saxena1 — APPROVED** [review]
Code quality, variable naming, UI contrast ratio, proper UI alignment and other things are taken care of properly.
Looks good to merge 👍

File: FitVid/src/Pages/Authentication/Login/Loginform.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import "../authentication.css";
+
+function Loginform() {
+  const [formDetails, setFormDetails] = useState({
+    email: "",
+    password: "",
+  });
+  const [isPasswordVisible, setPasswordVisible] = useState(false);
+  const { dispatchToast } = useToast();
+  // const { user, dispatchUser } = useUser();
```
Comment (gauravsinhaweb) [inline FitVid/src/Pages/Authentication/Login/Loginform.js:24]: remove commented code.

File: FitVid/src/Pages/Authentication/Signup/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+
+import "../authentication.css";
+
```
Comment (gauravsinhaweb) [inline FitVid/src/Pages/Authentication/Signup/SignUp.js]: remove extra spacing for visibility

File: FitVid/src/Pages/Authentication/AuthenticationBackground.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import "./authentication.css";
+function AuthenticationBackground() {
+  return (
+    <>
+      <img src="images/blob.svg" className="blob blob1" alt="top-left-blob" />
```
Comment (Shivani-1524) [inline FitVid/src/Pages/Authentication/AuthenticationBackground.js:6]: suggestion: classname blob blob1 could be better

File: FitVid/src/Pages/Authentication/AuthenticationBackground.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+      <img
+        src="images/blob.svg"
+        className="blob blob2"
```
Comment (Shivani-1524) [inline FitVid/src/Pages/Authentication/AuthenticationBackground.js:9]: same applies here

File: FitVid/src/Pages/Authentication/ForgotPassword/ForgotPassword.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+function ForgotPassword() {
+  return (
+    <div>
+      <Navbar />
+      <div className="spacer-3"></div>
+      <div className="wrapper">
+        <AuthenticationBackground />
+        <div className="authentication login">
+          <div className="authentication-image-container">
+            <img src="images/fitvid-forgot-password.svg" alt="signup-image" />
```
Comment (Shivani-1524) [inline FitVid/src/Pages/Authentication/ForgotPassword/ForgotPassword.js]: you dont have to mention image in the alt tag as screen readers read it with image added by default

File: FitVid/src/Pages/Authentication/Login/Loginform.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        type: "ADD_TOAST",
+        payload: { value: { id: uuid(), title: message, type: "SUCCESS" } },
+      });
+      navigate("/");
+    } else {
+      dispatchToast({
+        type: "ADD_TOAST",
+        payload: { value: { id: uuid(), title: message, type: "ERROR" } },
+      });
+      console.log(message);
```
Comment (Shivani-1524) [inline FitVid/src/Pages/Authentication/Login/Loginform.js:73]: could remove console.log

**Review by Shivani-1524 — APPROVED** [review]
Just a couple of suggestions, else good to merge.

File: FitVid/src/Pages/Authentication/Signup/SignUp.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+function SignUp() {
+  return (
+    <div>
+      <Navbar />
+      <div className="spacer-3"></div>
+      <div className="wrapper">
+        <AuthenticationBackground />
+        <div className="authentication">
+          <div className="authentication-image-container">
+            <img src="images/fitvid-signup.svg" alt="signup-image" />
```
Comment (Priya-Patel31) [inline FitVid/src/Pages/Authentication/Signup/SignUp.js:22]: in alt tag signup is enough no need to use image .

File: FitVid/src/Pages/Authentication/authentication.css [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+    margin: 1.8rem 0rem;
+}
+.form-input-box input{
+    width: 90%;
+    padding-left: 1rem;
+    font-size: 1rem;
+    height: 3.12rem;
+    box-shadow: rgba(0, 0, 0, 0.1) 0px 4px 12px;
+    border-radius: 3px;
+    border: 2px solid #BCBCBC;
```
Comment (Priya-Patel31) [inline FitVid/src/Pages/Authentication/authentication.css:81]: use variables for storing colors.

**Review by gauravsinhaweb — APPROVED** [review]
Good to merge after changes. some minor changes I mentioned, good use of common exports tough. Nice work!

File: FitVid/src/Util/loginUser.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import Axios from "axios";
+const loginUser = async (email, password) => {
+  console.log(email, password);
```
Comment (Priya-Patel31) [inline FitVid/src/Util/loginUser.js]: remove this console.log

File: FitVid/src/Util/loginUser.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+  try {
+    const response = await Axios.post("/api/auth/login", {
+      email: email,
+      password: password,
+    });
+
+    console.log(response);
```
Comment (Priya-Patel31) [inline FitVid/src/Util/loginUser.js]: same as above

File: FitVid/src/Util/loginUser.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    const token = response.data.encodedToken;
+    localStorage.setItem("token", token);
+
+    return {
+      data: response.data.foundUser,
+      success: true,
+      message: "Login Successfull",
```
Comment (Priya-Patel31) [inline FitVid/src/Util/loginUser.js:15]: spelling of successful is wrong.

File: FitVid/src/Util/signupUser.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+      password: password,
+    });
+
+    const token = response.data.encodedToken;
+    localStorage.setItem("token", token);
+
+    return {
+      data: response.data.createdUser,
+      success: true,
+      message: "SignUp Successfull",
```
Comment (Priya-Patel31) [inline FitVid/src/Util/signupUser.js:17]: same as above .

File: FitVid/src/Pages/Authentication/authentication.css [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+    display: block;
+    width: 90%;
+    margin: 1rem auto;
+    border-radius: 3px;
+    text-align: center;
+}
+.login{
+    padding: 0;
+}
+.forgot-password{
```
Comment (Priya-Patel31) [inline FitVid/src/Pages/Authentication/authentication.css:126]: could be written as .login,.forgot-password{  padding : 0 } as they have same properties.

File: FitVid/src/Pages/Authentication/authentication.css [hunk 14]
```diff
@@ -0,0 +1,162 @@
+.authentication{
+    display: grid;
+    grid-template-columns: 1fr 1fr;
+    /* background-color: var(--light-blue); */
```
Comment (Priya-Patel31) [inline FitVid/src/Pages/Authentication/authentication.css]: remove this commented code .

**Review by Priya-Patel31 — APPROVED** [review]
The code is well formatted and have taken care of best practices . Really liked the toasts which you have created and the way they are stacking . Just have suggested few minor changes .

File: FitVid/src/Pages/Authentication/Login/Loginform.js (hunk 1 above)
Comment (Ankur9669) [reply FitVid/src/Pages/Authentication/Login/Loginform.js:24]: This i will use it when i will integrate api calls

File: FitVid/src/Pages/Authentication/Signup/SignUp.js (hunk 2 above)
Comment (Ankur9669) [reply FitVid/src/Pages/Authentication/Signup/SignUp.js]: Done

File: FitVid/src/Pages/Authentication/Signup/SignUp.js (hunk 7 above)
Comment (Ankur9669) [reply FitVid/src/Pages/Authentication/Signup/SignUp.js:22]: Done

File: FitVid/src/Pages/Authentication/authentication.css (hunk 8 above)
Comment (Ankur9669) [reply FitVid/src/Pages/Authentication/authentication.css:81]: Yes but for box-shadow i am using like this only😅

File: FitVid/src/Util/loginUser.js (hunk 9 above)
Comment (Ankur9669) [reply FitVid/src/Util/loginUser.js]: Done

File: FitVid/src/Util/loginUser.js (hunk 10 above)
Comment (Ankur9669) [reply FitVid/src/Util/loginUser.js]: Done

File: FitVid/src/Util/loginUser.js (hunk 11 above)
Comment (Ankur9669) [reply FitVid/src/Util/loginUser.js:15]: It is right😅

File: FitVid/src/Pages/Authentication/authentication.css (hunk 14 above)
Comment (Ankur9669) [reply FitVid/src/Pages/Authentication/authentication.css]: Done

**Pull request merged**